update-nft-metadata-in-treasury/index.js

```diff
@@ -0,0 +1,93 @@
+import { PrivateKey, Client, TokenCreateTransaction, TokenType, TokenMintTransaction, TokenUpdateNftsTransaction } from "@hashgraph/sdk";
+import dotenv from 'dotenv';
+
+// Configure the environment variables
+dotenv.config();
+
+const operatorId = process.env.OPERATOR_ID;
+const operatorKey = process.env.OPERATOR_KEY;
+
+const client = Client.forPreviewnet().setOperator(operatorId, operatorKey);
+
+//Create a supply key to assign to the token
+const supplyKey = PrivateKey.generate();
+const supplyPublicKey = supplyKey.publicKey;
+
+async function createNftCollection() {
+    //Create an NFT with the supply key
+    const token = new TokenCreateTransaction()
+        .setTokenType(TokenType.NonFungibleUnique)
+        .setTokenName("MusicEvent2024")
+        .setTokenSymbol("MEVNT")
+        .setTreasuryAccountId(operatorId)
+        .setSupplyKey(supplyPublicKey)
+        .freezeWith(client);
+
+    const signTx3 = token.sign(supplyKey);
+    const submitTx3 = (await signTx3).execute(client);
+    const tokenReceipt = (await submitTx3).getReceipt(client);
+
+    // This is the nft collection Id
+    const tokenId = (await tokenReceipt).tokenId;
+    console.log(`The NFT collection ID is ${tokenId}`);
+
+    return tokenId;
+}
+
+async function mintTokenWithMetadata(tokenId) {
+    /*
+        Metadata should follow the JSON schema V2. 
+        This schema requires you to store your token metadata using a storage service.
+        View HIP-412 to learn more https://hips.hedera.com/hip/hip-412
+    */
+    const metadata = new Uint8Array([1]); // This should be a URI string [protocol://resource_location] or CID
```

Comment:
🤔 I think while this works at a "it runs" level, it doesn't really fit the *intent* of the HIP.

I suggest that perhaps you find a couple of files of IPFS (or any other content-addressable protocols that the HIP allows), and use those values as the initial and final metadata values, in order to stay true to the intent.